### ISSUE TITLE
DebuggerBreakpoints Not Hit in Debugger

### ISSUE THREAD

**Issue by ezolotko** [Microsoft/ClearScript#38]
Hi,

Here is the problem I experiencing with debugging a V8 script. Steps to reproduce:

1. Setup VS Code debugger with the following launch config (as described here https://microsoft.github.io/ClearScript/Details/Build.html, section V):
```
    "launch": {
        "version": "0.2.0",
        "configurations": [
            {
                "name": "Attach to ClearScript V8 on port 9222",
                "type": "node",
                "request": "attach",
                "protocol": "inspector",
                "address": "localhost",
                "port": 9222
            }
        ]
    }
```
2. Download the attached project:
[ClearScriptTest.zip](https://github.com/Microsoft/ClearScript/files/1621981/ClearScriptTest.zip)
open the solution in VS2017, restore packages, and run the application. As a result, a console with numbers running endlessly will open.

3. Attach VS Code debugger to the script. After the debugger is attached, click the pause button on VS Code debugger toolbar. Execution stops on the while(true) line:
![image](https://user-images.githubusercontent.com/993271/34818902-d0ed21f6-f6ee-11e7-8c41-860c269ba906.png)

4. Set breakpoints on any line inside the loop, and continue execution.
Expected result:
Execution stops on breakpoint hit.
Actual result:
Execution continues as if no breakpoints were set

OR

4. Click Step Over on the debugger toolbar.
Expected result:
Execution steps to the next line.
Actual result:
Execution continues without stopping.

Thank you.

**Comment by ClearScriptLib** [comment]
Hi @ezolotko,

This appears to be a bug in V8 6.2. After several iterations of a short loop, the script engine performs an optimizing transformation that effectively folds the loop into a single line, making it impossible to pause execution anywhere within the loop.

The V8 team seems to have fixed this in V8 6.3. Consider upgrading to [ClearScript 5.5.1.1](https://github.com/Microsoft/ClearScript/releases/tag/5.5.1.1) or later. The NuGet package you're using hasn't been updated, so you might want to [contact its owners](https://www.nuget.org/packages/ClearScript.V8/ContactOwners) or [build ClearScript yourself](https://microsoft.github.io/ClearScript/Details/Build.html).

Good luck!

**Comment by ezolotko** [comment]
Hi, I built 5.5.1.1, it fixes the problem. Thank you!
